### PR TITLE
Add test variable filter

### PR DESCRIPTION
**Pull request by peetya**
https://oat-sa.atlassian.net/browse/MS-88

### CODE REVIEW

**Comment by peetya** [comment]
> Well done!
> However,
> extractTestVariables() function already sort variables, what about to add a third parameter (null by default?) with the $filterSubmission to not sort them twice ? :D

Thanks! fixed :)

File: model/ResultsService.php [hunk 1]
```diff
@@ -848,29 +848,45 @@ public function getVariableDataFromDeliveryResult($resultIdentifier, $wantedType
         return $this->extractTestVariables($this->getVariablesFromObjectResult($testCallIds), $wantedTypes);
     }
 
-    public function extractTestVariables($variableObjects, $wantedTypes)
+    public function extractTestVariables(array $variableObjects, array $wantedTypes, string $filter = self::VARIABLES_FILTER_ALL)
```
Comment (julien-sebire) [inline model/ResultsService.php]: Are we ready to deal with scalar type hints in tao? Did you run tests against this code?

File: model/ResultsService.php [hunk 2]
```diff
@@ -848,29 +848,45 @@ public function getVariableDataFromDeliveryResult($resultIdentifier, $wantedType
...
+
+        usort($variableObjects, static function (
+            taoResultServer_models_classes_Variable $a,
+            taoResultServer_models_classes_Variable $b
+        ) use ($filter) {
+            if ($filter === self::VARIABLES_FILTER_LAST_SUBMITTED) {
+                return tao_helpers_Date::getTimeStamp($b->getEpoch()) - tao_helpers_Date::getTimeStamp($a->getEpoch());
             }
+
+            return tao_helpers_Date::getTimeStamp($a->getEpoch()) - tao_helpers_Date::getTimeStamp($b->getEpoch());
```
Comment (jbout) [inline model/ResultsService.php]: please use $a->getCreationTime() instead of tao_helpers_Date::getTimeStamp($a->getEpoch())

Comment (peetya) [reply model/ResultsService.php]: thanks, fixed

**Comment by peetya** [comment]
> Existing unit tests fail, but so does the develop branch...

If it's okay for you, let's fix it in another PR, to be able to merge this modification ASAP

**Comment by jbout** [comment]
Deal, as long as the new functionality gets covered as well ^^

**Comment by jbout** [comment]
One last comment: this is a new functionality so version should be 8.3.0 instead of 8.2.3

File: model/ResultsService.php (hunk 1 above)
Comment (peetya) [reply model/ResultsService.php]: yes, tests are passing

**Pull request merged**